core/storage/user/gae_models.py

```diff
@@ -97,6 +97,95 @@ def get_by_normalized_username(cls, normalized_username):
             cls.normalized_username == normalized_username).get()
 
 
+class ActivitiesCompletedByLearnerModel(base_models.BaseModel):
+    """Keeps track of all the explorations and collections completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    # IDs of all the explorations completed by the user.
+    completed_exploration_ids = ndb.StringProperty(repeated=True, indexed=True)
+    # IDs of all the collections completed by the user.
+    completed_collection_ids = ndb.StringProperty(repeated=True, indexed=True)
+
+
+class IncompleteCollectionsModel(base_models.BaseModel):
+    """Keeps track of all the collections currently being completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    incomplete_collection_ids = ndb.StringProperty(
+        repeated=True, indexed=True)
+
+
+class IncompleteExplorationsModel(base_models.BaseModel):
+    """Keeps track of all the explorations currently being completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    # The ids of the explorations partially completed by the user.
+    incomplete_exploration_ids = ndb.StringProperty(repeated=True, indexed=True)
+
+
+class ExpUserLastPlaythroughModel(base_models.BaseModel):
+    """Stores the "last playthrough" information of the exploration since the
+    user left it partially completed.
+
+    Instances of this class have keys of the form
+    [user_id].[exploration_id]
+    """
+    # The user id.
+    user_id = ndb.StringProperty(required=True, indexed=True)
+    # The exploration id.
+    exploration_id = ndb.StringProperty(required=True, indexed=True)
+    # The version of the exploration last played by the user.
+    version_last_played = ndb.IntegerProperty(default=None)
```

Comment:
last_exp_version_played (to match last_state_played)

edit: or maybe last_played_exp_version (to match last_played_state_name)